File: OOAD - Game Of Bricks/Player.java
/************************************************************************************/
import java.awt.*;
import java.awt.event.*;
import javax.swing.*;
/************************************************************************************/
class Player implements CONSTANTS
{
	private static Rectangle infoBorder;
	private static Color borderColor = Color.white;
	private static Color textColor = Color.white;
	private String name;
	private int score;
	private int lives;
/*==========================================================================*/	
	Player(Dimension dim, String name)
	{
		infoBorder = new Rectangle(10, 10, dim.width-30, 90);
		this.name = name;
		score = 0;
		lives = MAX_LIVES;
	}
/*==========================================================================*/
	public String getName() {return name;}
	public void setName(String n)
	{
		name = n;
	}
	public int getScore() {return score; }
	public void setScore(int s)
	{
		score = s;
	}
	public void setLives (int l)
	{
		lives = l;
	}
	public int getLives() {return lives;}
/*==========================================================================*/
	// it displays the Player's information
	public void displayPlayerInfo(Graphics g, int levelNo)
	{
		g.setColor(borderColor);
		g.drawRect(infoBorder.x, infoBorder.y, infoBorder.width, infoBorder.height);
		g.setColor(Color.yellow);
		g.setFont(new Font("ARIAL", 1, 17));
		
		g.drawString("Player Name  : " + name,  infoBorder.x+10, infoBorder.y+20);
		g.drawString("Score            : " + score, infoBorder.x+10, infoBorder.y+40);
		g.drawString("No. of lives    : " + lives, infoBorder.x+10, infoBorder.y+60);
		g.drawString("Level No.       : " + levelNo, infoBorder.x+10, infoBorder.y+80);

		// displays the key-board shortcuts for various options
		g.drawString("p : Pause Game", infoBorder.width-150, infoBorder.y+20);
		g.drawString("q : Quit Game", infoBorder.width-150, infoBorder.y+40);
		g.drawString("n : Next Level", infoBorder.width-150, infoBorder.y+60);
		g.drawString("b : Previous Level", infoBorder.width-150, infoBorder.y+80);

		// displays the title "GAME OF BRICKS"
		g.setColor(Color.green);
		g.setFont(new Font("ARIAL", 1, 50));
		g.drawString("GAME OF BRICKS", infoBorder.width/2 - 200, infoBorder.y+65);
	}
/*==========================================================================*/
	public void updateScore(int s)
	{
		score += s;
	}
	public void reduceLives()
	{
		lives--;
	}
	public boolean hasLife()
	{
		return (lives > 1);
	}
} // Player
/************************************************************************************/

File: PCCP - Game Of Bricks/src/player.cc
#include "player.h"
#include "header.h"
#include <string.h>
Player :: Player ()
{
	strcpy (name, DFT_NAME);
	score = 0;
	lives = 3;
}
int Player :: get_score()
{
	return score;
}
void Player :: updateScore (int s)
{
	score += s;
}
void Player :: set_score (int s)
{
	score = s;
}
int Player :: get_lives()
{
	return lives;
}
void Player :: set_lives (int l)
{
	lives = l;
}

void Player :: reduceLives ()
{
	lives--;
}
void Player :: readName()
{
	if (strcmp (name, DFT_NAME) == 0)
	{
		printw ("Enter your name : ");
		scanw  ("%s", name);
	}
}
/*
char& Player :: get_name()
{
	return *name;
}
*/

void Player :: set_name (char *nm)
{
}

int Player :: hasLife()
{
	return (lives > 1);
}
void Player :: displayInfo()
{
	mvprintw (0,0, "Player : %s", name);
	mvprintw (0,COLS-20, "p : Pause, q : Quit");
	mvprintw (1,0, "Score : %d", score);
	mvprintw (1, COLS-20, "Lives Remaining : %d", lives);
		for (int i=0; i<COLS; i++)
			mvaddch(TOP_EDGE-1,i,ACS_HLINE);
	refresh();
}



